packages/core/src/entityStatement/fetchEntityStatementChain.ts

```diff
@@ -0,0 +1,65 @@
+import type { EntityConfigurationClaims } from '../entityConfiguration'
+import { ErrorCode } from '../error/ErrorCode'
+import { OpenIdFederationError } from '../error/OpenIdFederationError'
+import type { VerifyCallback } from '../utils'
+import type { EntityStatementClaims } from './entityStatementClaims'
+import { fetchEntityStatement } from './fetchEntityStatement'
+
+export type FetchEntityStatementChainOptions = {
+  entityConfigurations: Array<EntityConfigurationClaims>
+  verifyJwtCallback: VerifyCallback
+}
+
+export const fetchEntityStatementChain = async ({
+  verifyJwtCallback,
+  entityConfigurations,
+}: FetchEntityStatementChainOptions) => {
+  // The trust anchors (i.e. the last item of the list) entity configuration will be used instead of a statement issued by a superior
+  const statements = [entityConfigurations[entityConfigurations.length - 1]]
```

Comment:
Do we check/ensure tbis array is non empty?

I think enabling noUncheckedIndexAccess or whatever it is called in TS compilerOptioms might be good

Reply:
Added it which did not help here (but some other places it did) so I just added a check manually that it is non-empty.